File: frontend/src/app/modules/contests/interfaces/contest.ts
import { ContestStatus } from "@app/modules/contests/constants/contest-status";

export class Contest {
  constructor(
    public id: number,
    public title: string,
    public description: string,
    public status: number,
    public problemsCount: number,
    public contestantsCount: number,
    public startTime: Date,
    public finishTime: Date,
    public logo: string,
    public category: number,
    public categoryTitle: string,
  ) {}

  isFinished(): boolean {
    return this.status === ContestStatus.FINISHED;
  }

  isAlready(): boolean {
    return this.status === ContestStatus.ALREADY;
  }

  isNotStarted(): boolean {
    return this.status === ContestStatus.NOT_STARTED;
  }
}

File: backend/apps/contests/serializers/contest.py
from django.db.models import Q
from rest_framework import serializers

from apps.contests.models import Contest


class ContestSerializer(serializers.ModelSerializer):
    contestants_count = serializers.IntegerField(source='contestants.count')
    problems_count = serializers.IntegerField(source='problems.count')
    category_title = serializers.CharField(source='get_category_display')
    status = serializers.IntegerField(source='get_status')

    class Meta:
        model = Contest
        fields = ['id', 'title', 'description', 'status',
                  'start_time', 'finish_time', 'category',
                  'category_title', 'logo',
                  'contestants_count', 'problems_count']
